user/src/main/java/org/the_chance/honeymart/ui/feature/product/ProductViewModel.kt

```diff
@@ -100,29 +103,29 @@ class ProductViewModel @Inject constructor(
     private fun getProductsByCategoryId() {
         _state.update { it.copy(isLoadingProduct = true, isError = false) }
         tryToExecute(
-            { getAllProducts(state.value.categoryId).map { it.toProductUiState() } },
+            { getAllProducts(state.value.categoryId) },
             ::onGetProductSuccess,
             ::onGetProductError
         )
     }
 
-    private fun onGetProductSuccess(products: List<ProductUiState>) {
+    private fun onGetProductSuccess(products: List<ProductEntity>) {
         if (products.isEmpty()) {
             _state.update {
                 it.copy(
                     isEmptyProducts = true,
-                    products = products
+                    products = products.map { it.toProductUiState() }
```

Comment:
The operation of mapping occurs three times in this view model and I think it should be replaced with one variable that holds the value of mapping so it can be assigned to any function that needs this type of mapping.